steem/utils.py

```diff
@@ -354,3 +363,77 @@ def strfdelta(tdelta, fmt):
 
 def is_valid_account_name(name):
     return re.match('^[a-z][a-z0-9\-.]{2,15}$', name)
+
+
+def compat_compose_dictionary(dictionary, **kwargs):
+    """
+    This method allows us the one line dictionary composition that is offered by the ** dictionary unpacking
+    available in 3.6.
+
+    :param dictionary: the dictionary to add the kwargs elements to.
+    :param kwargs: a set of key/value pairs to add to `dictionary`.
+    :return: the composed dictionary.
+    """
+    for pairs in kwargs.items():
+        dictionary[pairs[0]] = pairs[1]
+
+    return dictionary
+
+
+def compat_bytes(item, encoding=None):
+    """
+    This method is required because Python 2.7 `bytes` is simply an alias for `str`. Without this method,
+    code execution would look something like:
+
+    class clazz(object):
+
+        def __bytes__(self):
+            return bytes(5)
+
+
+    Python 2.7:
+
+    c = clazz()
+    bytes(c)
+    >>'<__main__.clazz object at 0x105171a90>'
+
+    In this example, when `bytes(c)` is invoked, the interpreter then calls `str(c)`, and prints the above string.
+    the method `__bytes__` is never invoked.
+
+    Python 3.6:
+    c = clazz()
+    bytes(c)
+    >>b'\x00\x00\x00\x00\x00'
+
+    This is the expected and necessary behavior across both platforms.
+
+    w/ future_bytes method, we will ensure that the correct bytes method is always invoked, avoiding the `str` alias in
+    2.7.
+
+    :param item: this is the object who's bytes method needs to be invoked
+    :param encoding: optional encoding parameter to handle the Python 3.6 two argument 'bytes' method.
+    :return: a bytes object that functions the same across 3.6 and 2.7
+    """
+    if hasattr(item, '__bytes__'):
+        return item.__bytes__()
+    else:
+        if encoding:
+            return bytes(item, encoding)
+        else:
+            return bytes(item)
```

Comment:
Do we need the `if encoding` branch or can we just send None as the 2nd arg in cases where it's left blank?

Reply:
Yes, although the logic is potentially obfuscated for simplicity:
`bytes(item)` is valid in both 2.7 and 3.6
`bytes(item, encoding)` is valid ONLY in 3.6.

This allows for the simplest decision tree.

Reply:
:+1: